cmd/theatre-consoles/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"context"
+	stdlog "log"
+	"os"
+
+	"github.com/alecthomas/kingpin"
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	theatre "github.com/gocardless/theatre/pkg/client/clientset/versioned"
+	"github.com/gocardless/theatre/pkg/logging"
+	"github.com/gocardless/theatre/pkg/signals"
+	"github.com/gocardless/theatre/pkg/workloads/console/runner"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp" // this is required to auth against GCP
+	"k8s.io/klog"
+	kconf "sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+var (
+	cli = kingpin.New("consoles", "Manages theatre consoles")
+
+	create = cli.Command("create", "Creates a new console given a template")
+)
+
+func main() {
+	// Set up logging
+	logger := kitlog.NewLogfmtLogger(os.Stderr)
+	logger = level.NewFilter(logger, level.AllowInfo())
+	logger = kitlog.With(logger, "ts", kitlog.DefaultTimestampUTC, "caller", logging.RecorderAwareCaller())
+	stdlog.SetOutput(kitlog.NewStdlibAdapter(logger))
+	klog.SetOutput(kitlog.NewStdlibAdapter(logger))
+
+	ctx, _ := signals.SetupSignalHandler()
+
+	if err := Run(ctx, logger); err != nil {
+		cli.Fatalf("unexpected error: %s", err)
+	}
+}
+
+func Run(ctx context.Context, logger kitlog.Logger) error {
+	config, err := kconf.GetConfig()
+	if err != nil {
+		return err
+	}
+
+	client, err := kubernetes.NewForConfig(config)
```

Comment:
Personally I'd like to have the kube context parameterisable here. Otherwise it becomes a bit awkward if you need to boot a console in two or more clusters, needing to switch your context around. WDYT?

Reply:
Reflecting on this, I think it would be useful to parametrise this.
It would also be useful in the case you wanted to script anything using this cli.